### PR TITLE
Fix #377: EE10 Faces open source libraries

### LINKED ISSUE #377: Jakarta EE10: Faces upgrade open source libraries
4 of the most popular open source libraries PrimeFaces, PrimeFaces Extensions, OmniFaces, and MyFaces upgrades.

EE8:
```xml
<dependency>
	<groupId>org.primefaces</groupId>
	<artifactId>primefaces</artifactId>
	<version>13.0.3</version>
</dependency>
<dependency>
	<groupId>org.primefaces.extensions</groupId>
	<artifactId>primefaces-extensions</artifactId>
	<version>13.0.3</version>
</dependency>
<dependency>
	<groupId>org.omnifaces</groupId>
	<artifactId>omnifaces</artifactId>
	<version>3.2</version>
</dependency>
<dependency>
	<groupId>org.apache.myfaces.core</groupId>
	<artifactId>myfaces-api</artifactId>
	<version>2.3.10</version>
</dependency>
<dependency>
	<groupId>org.apache.myfaces.core</groupId>
	<artifactId>myfaces-impl</artifactId>
	<version>2.3.10</version>
</dependency>
```

EE10 (PF gets jakarta classifier, and OmniFaces and MyFaces upgrades):

```xml
<dependency>
	<groupId>org.primefaces</groupId>
	<artifactId>primefaces</artifactId>
	<version>13.0.3</version>
	<classifier>jakarta</classifier>
</dependency>
<dependency>
	<groupId>org.primefaces.extensions</groupId>
	<artifactId>primefaces-extensions</artifactId>
	<version>13.0.3</version>
	<classifier>jakarta</classifier>
</dependency>
<dependency>
	<groupId>org.omnifaces</groupId>
	<artifactId>omnifaces</artifactId>
	<version>4.3</version>
</dependency>
<dependency>
	<groupId>org.apache.myfaces.core</groupId>
	<artifactId>myfaces-api</artifactId>
	<version>4.0.1</version>
</dependency>
<dependency>
	<groupId>org.apache.myfaces.core</groupId>
	<artifactId>myfaces-impl</artifactId>
	<version>4.0.1</version>
</dependency>
```

### PR DESCRIPTION
**Pull request by melloware**
Fix #377: EE10 Faces open source libraries

### CODE REVIEW

File: src/main/resources/META-INF/rewrite/jakarta-faces-4.yml [hunk 1]
```diff
@@ -448,3 +449,77 @@ recipeList:
...
+  - org.openrewrite.maven.ChangeDependencyClassifier:
+      groupId: org.primefaces
+      artifactId: primefaces
+      newClassifier: jakarta
+  - org.openrewrite.maven.ChangeDependencyGroupIdAndArtifactId:
+      oldGroupId: org.primefaces
+      oldArtifactId: primefaces
+      newGroupId: org.primefaces
+      newArtifactId: primefaces
+      newVersion: 13.0.3
```
Comment (timtebeek) [inline src/main/resources/META-INF/rewrite/jakarta-faces-4.yml:467]: Might make sense to [use Xrange version selectors here](https://docs.openrewrite.org/reference/dependency-version-selectors#xrange), such that we don't pin folks to any particular version, but for instance anything in `13.x` or `13.0.x`

Comment (melloware) [reply src/main/resources/META-INF/rewrite/jakarta-faces-4.yml]: Great feedback updated them all to xrange!

File: src/main/resources/META-INF/rewrite/jakarta-faces-4.yml [hunk 2]
```diff
@@ -448,3 +449,77 @@ recipeList:
...
+      oldGroupId: org.primefaces
+      oldArtifactId: primefaces
+      newGroupId: org.primefaces
+      newArtifactId: primefaces
+      newVersion: 13.0.x
+  - org.openrewrite.maven.ChangeManagedDependencyGroupIdAndArtifactId:
+      oldGroupId: org.primefaces
+      oldArtifactId: primefaces
+      newGroupId: org.primefaces
+      newArtifactId: primefaces
```
Comment (timtebeek) [inline src/main/resources/META-INF/rewrite/jakarta-faces-4.yml:472]: There's no need to repeat the groupId or artifactId when the value remains the same. Just the old values is then enough. Sorry I didn't spot that the first time around!

Comment (timtebeek) [reply src/main/resources/META-INF/rewrite/jakarta-faces-4.yml:472]: Also: you're using the Maven variant of the recipe, whereas in. Rewrite-java-dependencies we also have one to change both Maven and gradle dependency versions

Comment (timtebeek) [reply src/main/resources/META-INF/rewrite/jakarta-faces-4.yml:472]: https://docs.openrewrite.org/recipes/java/dependencies/upgradedependencyversion

Comment (melloware) [reply src/main/resources/META-INF/rewrite/jakarta-faces-4.yml:472]: Ok let me look at that one you mentioned

Comment (melloware) [reply src/main/resources/META-INF/rewrite/jakarta-faces-4.yml:477]: OK I updated to the UpgradeDependencyVersion but for ChangedManagedVersion it does NOT work without the `new` values it says they are required and errors.

Also I am updating the Classifier to `jakarta` but in main rewrite there is no way to change a "Managed Dependency" classifier it looks like that Recipe is missing?

Comment (melloware) [reply src/main/resources/META-INF/rewrite/jakarta-faces-4.yml:477]: I opened this ticket: https://github.com/openrewrite/rewrite/issues/3830

**Review by timtebeek — APPROVED** [review]

**Comment by timtebeek** [comment]
Thanks again! I've removed some duplicate recipes and added a test. Should be good to go ...

... but we need this fix to merge first and then update the branch here to get the tests to pass.
- https://github.com/openrewrite/rewrite-migrate-java/pull/379

**Pull request merged**